### PR TITLE
Convert table dtype to dtype of merge table before appending, fixes 671

### PR DESCRIPTION
**Pull request by maxnoe**
@morcuended Can you try this out?

### CODE REVIEW

**Comment by morcuended** [comment]
Sure, let me try.

File: lstchain/scripts/lstchain_merge_hdf5_files.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def main():
 
     if args.noimage:
         keys = get_dataset_keys(file_list[0])
-        keys = [k for k in keys if 'image' not in k]
+        keys = {k for k in keys if 'image' not in k}
```
Comment (morcuended) [inline lstchain/scripts/lstchain_merge_hdf5_files.py:79]: Is this still needed given that it is later converted to `set()`?

Comment (maxnoe) [reply lstchain/scripts/lstchain_merge_hdf5_files.py:79]: Not strictly needed, but if keys is expected to be a set why not make it a set directly?

**Review by morcuended — APPROVED** [review]
Thanks @maxnoe. It worked fine with the fix. I tried with DL2/DL1, with/without `--smart` and `--no-image` option

**Comment by maxnoe** [comment]
Thanks for checking

**Pull request merged**